panel-protocol/src/lib.rs

```diff
@@ -165,16 +167,23 @@ impl ReportReader {
         Self { buf: ArrayVec::new() }
     }
 
-    pub fn process_byte(&mut self, byte: u8) -> Result<Option<Report>, Error> {
-        self.buf.try_push(byte).map_err(|_| Error::BufferFull)?;
-        match Report::try_from(&self.buf[..]) {
-            Ok(Some((command, bytes_read))) => {
-                self.buf.drain(0..bytes_read);
-                Ok(Some(command))
-            },
-            Err(_) => Err(Error::MalformedMessage),
-            Ok(None) => Ok(None),
+    pub fn process_bytes(&mut self, bytes: &[u8]) -> Result<ArrayVec<[Report; MAX_REPORT_QUEUE_LEN]>, Error> {
+        self.buf.try_extend_from_slice(bytes).map_err(|_| Error::BufferFull)?;
+
+        let mut output = ArrayVec::new();
+
+        loop {
+            match Report::try_from(&self.buf[..]) {
+                Ok(Some((report, bytes_read))) => {
+                    self.buf.drain(0..bytes_read);
+                    output.push(report);
```

Comment:
check if `output.len()` is less than `MAX_REPORT_QUEUE_LEN`, and otherwise report an error instead of panicing?